### PR TITLE
wine-unstable: 1.9.7 -> 1.9.11; add optional dependencies; opencl-icd: init; refactor flags

### PR DESCRIPTION
**Pull request by bendlas**
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/misc/emulators/wine/base.nix [hunk 1]
```diff
@@ -22,10 +22,16 @@ stdenv.mkDerivation ((lib.optionalAttrs (! isNull buildScript) {
   buildInputs = toBuildInputs pkgArches (pkgs: (with pkgs; [
```
Comment (abbradar) [inline pkgs/misc/emulators/wine/base.nix]: Minor nitpick: you init `opencl-icd` in another commit, so this commit message is confusing.

**Comment by abbradar** [comment]
I'm somewhat reluctant on adding all those dependencies to Wine. How much people would really use its, say, ODBC or GSM bindings? IMHO most of them won't matter to the vast majority of users if any, while blowing up the closure size.

Having said that, some people might actually find this useful. I propose to go with one addition at a time, backed by a use case. When the mass reaches critical levels we can refactor things to add `with*` flags so that people would get what they want with recompilation. What do you think?

**Comment by avnik** [comment]
On Tue, May 10, 2016 at 02:52:53PM -0700, Nikolay Amiantov wrote:

> Having said that, some people might actually find this useful.
> I propose to go with one addition at a time, backed by a use
> case. When the mass reaches critical levels we can refactor
> things to add `with*` flags so that people would get what they
> want with recompilation. What do you think?
> 

I am agreed with it. May be with couple presets, like "wineFull".
Also wineStaging can get some customization.

(anyway most games usually stick with "wine version, that work
 stable with that game", so rebuilds is unavoidable ;))

File: pkgs/misc/emulators/wine/base.nix (hunk 1 above)
Comment (abbradar) [inline pkgs/misc/emulators/wine/base.nix]: Nevermind; I've confused title of the PR with commit message in the GitHub's UI.

**Comment by bendlas** [comment]
I pushed another patch, to encode the (already depended on) libtxc_dxtn library into RPATH.
Regarding closure size:

```
> du -sh --total $(nix-store -qR /nix/store/8mynl7igi0s3zvg73b5d4hkfj4wkjzqw-wine-wow-1.9.8-staging)
...
1.5G    total
```
vs
```
> du -sh --total $(nix-store -qR /nix/store/0f88afn3fn4zxwwpsg7vzb0j3dyvqql9-wine-wow-1.9.9-staging)
...
1.8G    total
```

If you feel that's too much, I can hide it behind flags

**Comment by abbradar** [comment]
That's indeed much but it also depends on whether what's gotten into closure is usually installed with Wine anyway or not. My guess is that  `glib` and `gtk3` may be the main space eaters (and certainly not all Wine users would have them). Try to disable them and see what happens -- if it indeed helps let's move them behind flags.

**Comment by bendlas** [comment]
I refactored wine's build expression to have a flag for everything, that's optional. Default flags are slightly less than in the original expression. I hope that this won't break any dependent, but it seems pretty reasonable, in that it includes widely used features like tls, curses, alsa, opengl and image formats.

**Comment by bendlas** [comment]
@abbradar is there anything missing on this?

**Comment by abbradar** [comment]
Completely forgot about this one, sorry! I'll review this once more, build and merge if there are no problems.

**Comment by abbradar** [comment]
I haven't built all combinations but at least `wineUnstable` builds and `wineFull` configures. Thank you!

**Pull request merged**